wikimedia/wikipedia-ios
# Fix table of contents scrolling

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T257369

### Notes
* This fixes a few things - 
- On a large screen, tapping a reference backlink or tapping the "jump to reference section" button updates an always-open Table of Contents.
- On all size devices, table of contents now update the highlight when moving up to the first section of the article via reference backlinks.
- On all size devices, table of contents should be scrolled appropriately to see the highlighted section when jumping around an article via reference links.

This does NOT fix a related bug - If scrolling through an article "About this article" and "Read more" will never be selected. (They will work if directly tapping table of contents.) This is because the iOS code calls them section -2 and -3 respectively, and the actual article HTML doesn't give any section number to these two sections. This could be fixed in the (local WWW) JS addFooter code. That said, this is a very corner case bug. On all but the smallest screen phones, the highlight naturally is the final article section when scrolled all the way to the bottom of the article - because "About this Article" and "Read More" don't use up an entire screen, and so a bit of the final section shows through. Given the small amount of cases and lack of side effects for these final two sections I'm not going to fix it right now.

### Test Steps
1. On a large-screen sized iPhone, use a reference to jump to the references section. Open Table of Contents, ensure it's scrolled to "References" section and that section is highlighted.
2. Tap a reference to scroll back up in the article. Open Table of Contents, ensure it's scrolled to appropriate section and that section is highlighted.
3. Rotate the phone so it's in landscape. Perform steps 1 and 2 with the TOC permanently on the screen.
4. Do this on an iPad, as well (optional).
